Documentation/network/concepts/masquerading.rst

```diff
@@ -163,10 +163,21 @@ option ``egress-masquerade-interfaces: eth0`` can be used.
    ``eth+``, all interfaces matching the prefix ``eth`` will be used for
    masquerading.
 
+
 For the advanced case where the routing layer would select different source addresses
 depending on the destination CIDR, the option ``enable-masquerade-to-route-source: "true"``
 can be used in order to masquerade to the source addresses rather than to the
 primary interface address. The latter is then only considered as a catch-all
 fallback, and for the default routes. For these advanced cases the user needs
 to ensure that there are no overlapping destination CIDRs as routes on the
 relevant masquerading interfaces.
+
+With the ``enable-masquerade-to-route-source: "true"`` option, Cilium will, 
+by default, use interfaces listed in the ``devices`` field as the egress masquerade interfaces 
+when ``egress-masquerade-interfaces`` is empty. When ``egress-masquerade-interfaces`` is set, 
+it takes precedence over ``devices`` to restrict which network interface should perform masquerading.
+
+.. note::
+
+   with ``enable-masquerade-to-route-source: "true"`` option ,
+   you can set egress-masquerade-interfaces as a list such as ``eth1 eth2 eth3``.
```

Comment:
From https://docs.cilium.io/en/stable/network/concepts/routing/#id6 it looks like we also support `eth+` which seems like a better suggestion?

```suggestion
   you can set egress-masquerade-interfaces as a list such as ``eth+``.
```

Reply:
the reason that I use this `eth1 eth2 eth3` format here in the doc for the following reasons:

1. `eth1 eth2 eth3` format is only supported when I enable the `enable-masquerade-to-route-source`(https://github.com/cilium/cilium/blob/main/pkg/datapath/iptables/iptables.go#L1207-L1222). Cilium will consider `eth1 eth2 eth3` as one interface with 1.16.4(https://github.com/cilium/cilium/blob/v1.16.4/pkg/datapath/iptables/iptables.go#L1318-L1321)
2. In real world use cases , I found the interfaces here are actually will not just be `eth`. Therefore, I feel this format will be more aligned with the config that we will put in the environment.

Reply:
> Cilium will consider eth1 eth2 eth3 as one interface with 1.16.4

Do you mean to say that Cilium 1.15.x has different behaviour than 1.16.y? This sounds like a bug in the usability for this flag, which makes me a bit nervous to document to users to use it one way or another until we agree on the right configuration format + fix the bugs.

Reply:
Sorry for the confusion. what I'm trying to say is the list format is supported with `enable-masquerade-to-route-source: true` from day 1 when we have the `enable-masquerade-to-route-source: true` option.

I tested this afternoon with 1.16.4 that when I dont have `enable-masquerade-to-route-source` enabled with `egress-masquerade-interfaces: eth1 eth2 eth3`, cilium is considering the list as one interface and can't install the iptables rules correctly. Then I noticed that we have https://github.com/cilium/cilium/pull/36103 to fix the issue from you. 

This PR is only describing people can use the list format when they enable `enable-masquerade-to-route-source: true`, so we can safely backport it to 1.15 and 1.16.

Hope I made it clear this time

Reply:
I see, makes sense. I appreciate you digging into this and explaining it clearly to me :pray: